Clic&Farm-iOS/Controllers/AddInterventionViewController.swift

```diff
@@ -0,0 +1,152 @@
+        interventionToolsTableView.layer.cornerRadius = 4
+        
+        // Crops select
+        validateButton.layer.cornerRadius = 3
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    //MARK: Table view
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return interventionTools.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        var cell = UITableViewCell()
+        
+        switch tableView {
+        case interventionToolsTableView:
+            cell = tableView.dequeueReusableCell(withIdentifier: "InterventionToolsTableViewCell", for: indexPath)
+        default:
+            return cell
+        }
+        
+        //cell.layoutMargins = UIEdgeInsets.zero
+        
+        return cell
+    }
+    
+    /*
+     // MARK: - Navigation
+     
+     // In a storyboard-based application, you will often want to do a little preparation before navigation
+     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+     // Get the new view controller using segue.destinationViewController.
+     // Pass the selected object to the new view controller.
+     }
+     */
+    
+    //MARK: Actions
+    
+    var animationRunning: Bool = false
+    
+    func animateView(isCollapse: Bool, angle: CGFloat) {
+        heightConstraint.constant = isCollapse ? 50 : 300
+        
+        animationRunning = true
+        
+        UIView.animate(withDuration: 0.5, animations: {
+            self.collapseButton.imageView!.transform = CGAffineTransform(rotationAngle: angle)
+            self.view.layoutIfNeeded()
+        }, completion: { _ in
+            self.animationRunning = false
+        })
+    }
+    
+    //MARK: - Actions
+    
+    @IBAction func selectCrops(_ sender: Any) {
+        self.dimView.isHidden = false
+        self.selectCropsView.isHidden = false
+        UIView.animate(withDuration: 1, animations: {
+            UIApplication.shared.statusBarView?.backgroundColor = UIColor.black
+        })
+    }
+    
+    @IBAction func validateCrops(_ sender: Any) {
+        self.dimView.isHidden = true
+        self.selectCropsView.isHidden = true
+        UIView.animate(withDuration: 1, animations: {
+            UIApplication.shared.statusBarView?.backgroundColor = UIColor(rgb: 0x175FC8)
+        })
+    }
+    
+    @IBAction func collapseExpand(_ sender: Any) {
+        if animationRunning {
+            return
+        }
+        
+        let shouldCollapse = firstView.frame.height != 50
+        
+        animateView(isCollapse: shouldCollapse, angle: shouldCollapse ? CGFloat.pi : CGFloat.pi - 3.14159)
```

Comment:
What's the use of this `CGFloat.pi - 3.14159`? Why not `0`?

Reply:
Magic trick to rotate back, didn't found any other solution

Reply:
What happens if you set `0` instead?

Reply:
The image rotates but without rotating back. It rotates 180 degrees but I want -180 degrees.
It's because the rotation is made taking the shortest way.

Reply:
So it goes around making a full circle instead of going one way and back along the same half-circle?

Reply:
No, that's the contrary. it goes one way and then goes back along the same half-circle.

Reply:
And we want it to go full circle?

Reply:
Yes, but that was just a first version of animations, to learn how to animate views.

Reply:
Alright then!